ApiTestingSolution/Tests/UserControllerTests.cs

```diff
@@ -187,6 +203,134 @@ public void UpdateUserByIncorrectRequestBodyTest(Method method)
+                Assert.That(users, Does.Not.Contain(userToDelete), $"{userToDelete} is not deleated");
+                Assert.That(codes, Does.Contain(userToDelete.ZipCode), $"Zip code: {userToDelete.ZipCode} should be avaailible");
+            });
+        }
+
+        [Test]
+        [AllureFeature("DeleteByRequiredFieldsOnlyTest")]
+        [AllureStory("Validate delete user using required fields only")]
+        public void DeleteByRequiredFieldsOnlyTest()
+        {
+            var userToDelete = UserControllerService.GetAllUsers().Users.FirstOrDefault();
+            var jsonBody = $"{{ \"name\": \"{userToDelete.Name}\", \"sex\": \"{userToDelete.Sex}\" }}";
+
+            var response = UserControllerService.DeleteUser(jsonBody);
+            var zipCodes = ZipCodeControllerService.GetAvailableZipCodes();
+            var codes = JsonHelper.DeserializeJsonContent<List<string>>(zipCodes);
+            var users = UserControllerService.GetAllUsers().Users;
+
+            Assert.Multiple(() =>
+            {
+                Assert.That(response.StatusCode, Is.EqualTo(HttpStatusCode.NoContent), "Status code should be 204 No Content");
+                Assert.That(users, Does.Not.Contain(userToDelete), $"{userToDelete} is not deleated");
+                Assert.That(codes, Does.Contain(userToDelete.ZipCode), $"Zip code: {userToDelete.ZipCode} should be avaailible");
+            });
+        }
+
+        [Test]
+        [AllureFeature("DeleteByMissedRequiredFieldTest")]
+        [AllureStory("Validate delete user without using required fields")]
+        public void DeleteByMissedRequiredFieldTest()
+        {
+            var userToDelete = UserControllerService.GetAllUsers().Users.FirstOrDefault();
+            var jsonBody = $"{{ \"age\": \"{userToDelete.Age}\", \"sex\": \"{userToDelete.Sex}\", \"zipCode\": \"{userToDelete.ZipCode}\" }}";
+            
+            var response = UserControllerService.DeleteUser(jsonBody);
+            var zipCodes = ZipCodeControllerService.GetAvailableZipCodes();
+            var codes = JsonHelper.DeserializeJsonContent<List<string>>(zipCodes);
+            var users = UserControllerService.GetAllUsers().Users;
+
+            Assert.Multiple(() =>
+            {
+                Assert.That(response.StatusCode, Is.EqualTo(HttpStatusCode.Conflict), "Status code should be 409 Conflict");
+                Assert.That(users, Does.Contain(userToDelete), $"{userToDelete} is deleated");
+            });
+        }
+
+        [Test]
+        [AllureFeature("UploadFileWithCorrectUsersDataTest")]
+        [AllureStory("Validate upload file with correct users data")]
+        public void UploadFileWithCorrectUsersDataTest()
+        {
+            var file = UserHelpers.CreateJsonFileWithCorrectUsers();
+            var usersFromFile = file.Users;
+            var usedCodes = usersFromFile.Select(user => user.ZipCode).ToList();
+            var filePath = file.FilePath;
```

Comment:
I think that you can you `file.FilePath` construction within your test directly without wrapping it into a separate variable with name `filePath`